### PR TITLE
Add support for plog using vcpkg

### PR DESCRIPTION
**Pull request by abeltrano**
### Type

- [ ] Bug fix
- [ ] Feature addition
- [ ] Feature update
- [ ] Breaking change
- [X] Non-functional change
- [ ] Documentation
- [X] Infrastructure

### Goals

Allow the use of the [plog](https://github.com/SergiusTheBest/plog) open source library, using `vcpkg` to resolve the dependency.

### Technical Details

* Restore local development configuration for `vcpkg`.
* Detect when the project is being configured on behalf of `vcpkg` to fix git dependency issue.

### Test Results

* Successfully built the simulator driver with local test commit.
* Successfully built complete repo using `vcpkg-dev` cmake preset.
* Successfully built the nearobject tree in Windows.

### Reviewer Focus

None

### Future Work

* The plog library must also be made available using `FetchContent`. ng with libevent has been completed.

### Checklist

- [ ] Build target `all` compiles cleanly.
- [ ] clang-format and clang-tidy deltas produced no new output.
- [ ] Newly added functions include doxygen-style comment block.

### CODE REVIEW

File: ports/nearobject-framework/vcpkg.json [hunk 1]
```diff
@@ -1,41 +1,42 @@
...
-      ],
-      "supports": "windows"
-    }
-  },
-  "builtin-baseline": "779ce74ef67d3e12d904da1b15f9ed5626d5f677"
-}
+{
+  "$schema": "https://raw.githubusercontent.com/microsoft/vcpkg-tool/main/docs/vcpkg.schema.json",
+  "name": "nearobject-framework",
+  "version-string": "0.0.2",
```
Comment (abeltrano) [inline ports/nearobject-framework/vcpkg.json:4]: Not sure why this whole file is being flagged as changed. The only changes are the version bump to 0.0.2 on this line, and the addition of "plog" on line 10.

File: CMakeLists.txt [hunk 2]
```diff
@@ -57,13 +72,15 @@ set(VERSION_NLOHMANN_JSON 3.11.2)
 set(VERSION_MAGIC_ENUM 0.8.1)
 set(VERSION_CLI11 2.2.0)
 set(VERSION_CATCH2 3.1.0)
+set(VERSION PLOG 1.1.9)
```
Comment (forwardpointer) [inline CMakeLists.txt:75]: should this be VERSION_PLOG?
```suggestion
set(VERSION_PLOG 1.1.9)
```

Comment (abeltrano) [reply CMakeLists.txt:75]: Yep, thanks!

Comment (abeltrano) [reply CMakeLists.txt:75]: Fixed in #50

**Pull request merged**